File: Projects/jyleo/src/org/leo/shell/widget/CutCopyPaste.java
//@+leo-ver=4-thin
//@+node:zorcanda!.20051117153808:@thin CutCopyPaste.java
//@@language java
package org.leo.shell.widget;

import java.awt.event.*;
import javax.swing.AbstractAction;
import javax.swing.JMenu;
import javax.swing.JPopupMenu;
import javax.swing.text.JTextComponent;

public class CutCopyPaste extends MouseAdapter{

    final JTextComponent jtc;
    public CutCopyPaste( JTextComponent jtc ){
    
        this.jtc = jtc;
        jtc.addMouseListener( this );
    
    
    }

    public void mousePressed( MouseEvent me ){
    
    
        if( me.getButton() == me.BUTTON3 ){
        
            JMenu jm = new JMenu();
            AbstractAction cut = new AbstractAction( "Cut" ){
            
                public void actionPerformed( ActionEvent ae ){
                
                    jtc.cut();
                
                }
            
            };
            AbstractAction copy = new AbstractAction( "Copy" ){
            
                public void actionPerformed( ActionEvent ae ){
                
                    jtc.copy();
                
                }
            
            
            };
            AbstractAction paste = new AbstractAction( "Paste" ){
            
                public void actionPerformed( ActionEvent ae ){
                    
                    jtc.paste();
                
                }
            
            
            };
            jm.add( cut );
            jm.add( copy );
            jm.add( paste );
            JPopupMenu jpm = jm.getPopupMenu();
            jpm.show( jtc, me.getX(), me.getY() );
        
        }
    
    
    
    }



}
//@-node:zorcanda!.20051117153808:@thin CutCopyPaste.java
//@-leo

File: Projects/jyleo/src/CutCopyPaste.py
#@+leo-ver=4-thin
#@+node:zorcanda!.20050804212319:@thin CutCopyPaste.py
import java
import javax.swing as swing
import java.awt.event as aevent


class CutCopyPaste( aevent.MouseAdapter ):
    
    def __init__( self, jtp ):
        aevent.MouseAdapter.__init__( self )
        self.jtp = jtp
        self.ccp = swing.JPopupMenu()
        cut = swing.JMenuItem( "Cut" )
        cut.actionPerformed = lambda event : jtp.cut()
        self.ccp.add( cut )
        copy = swing.JMenuItem( "Copy" )
        copy.actionPerformed = lambda event : jtp.copy()
        self.ccp.add( copy )
        paste = swing.JMenuItem( "Paste" )
        paste.actionPerformed = lambda event: jtp.paste()
        self.ccp.add( paste )
        jtp.addMouseListener( self )
    
    
    def _paste( self ):
        
        dtk = java.awt.Toolkit.getDefaultToolkit()
        cb = dtk.getSystemClipboard()
        df = cb.getAvailableDataFlavors()
        for z in df:
            print z
        
       
    def mousePressed( self, event ):
        
        if event.getButton() == event.BUTTON3:
            self.summon_ccp( event )
            
    
    def summon_ccp( self, event ):
        
        self.ccp.show( event.getSource(), event.getX(), event.getY() )
#@-node:zorcanda!.20050804212319:@thin CutCopyPaste.py
#@-leo
